components/rmrk/Gallery/Holder/Holder.vue

```diff
@@ -345,37 +366,61 @@ export default class Holder extends mixins(ChainMixin, KeyboardEventsMixin) {
     return Object.values(itemRowMap)
   }
 
-  private generateHolderGroups(itemRowList): TableRow[] {
-    const holderGroups: Record<string, TableRow> = {}
-    itemRowList
-      .filter((item) => item.Holder !== '-')
-      .forEach((item) => {
-        const holder = item['Holder']
-        if (holderGroups[holder]) {
-          holderGroups[holder].Items.push(item)
-          holderGroups[holder]['Bought'] =
-            holderGroups[holder]['Bought'] + item['Bought']
-          holderGroups[holder]['Sale'] =
-            holderGroups[holder]['Sale'] + item['Sale']
-        } else {
-          holderGroups[holder] = {
-            ...item,
-            Items: [item],
-          }
+  private getGroupNameFromRow(item: TableRow): string {
+    if (this.groupKey === 'Holder') {
+      return item['Holder']
+    } else if (this.groupKey === 'CollectionId') {
+      return item['Item']['collection']['id']
+    }
+    return item['Holder']
```

Comment:
small refacto

```suggestion
		return this.groupKey !== 'Holder'
			? item['Item']['collection']['id']
			: item['Holder']
```

Reply:
It seems that it don't need to change, because we may add more group type in the future, not just 'Holder' and 'CollectionId'.

Reply:
Alright, in that case, for later, I would suggest to use a switch case or an object lookups to avoid having a long if...else statements